Islandora-CLAW/islandora
# Related link header

**GitHub Issue**: Part of https://github.com/Islandora-CLAW/CLAW/issues/716

Other Relevant Links:
- https://github.com/Islandora-CLAW/islandora/pull/68 needs to get merged first.
- https://github.com/Islandora-CLAW/islandora_collection/pull/24 adds the collection header code back in

# What does this Pull Request do?

Adds a `rel="related"` link header to GET and HEAD response for each entity referenced by an entity reference field.  The `title` of the link header is the field's label.  So for a resource that has a tiff, a lower res display copy, and belongs to a collection, I got: 

```
<http://localhost:8000/node/2>; rel="related"; title="Member Of", <http://localhost:8000/media/2>; rel="related"; title="Tiff image", <http://localhost:8000/media/3>; rel="related"; title="Web Image"
```

I've moved the collection specific link header to the islandora_collection module in another pull request.

# What's new?
* A new `View Alter` interface
* A base class for view alters that add link headers
* A view alter that adds the related link headers, exposed as a service
* Functional tests 

# How should this be tested?

Go make an Islandora Image and then hit it with a GET or HEAD request.   You should see link headers for all the entities you made though the form.  They follow the pattern of `<node_uri>; rel="related"; title="Field Label"`  

# Interested parties
@Islandora-CLAW/committers
